File: kotlin/src/main/kotlin/utils/TextFileProcessor.kt
package utils

import java.io.File

class TextFileProcessor(private val year: Int, private val dayNumber: Int) {

    fun getFileLines(): List<String> {

        val fileLines = ArrayList<String>()

        val fileLocation = "src/main/resources/$year/day$dayNumber.txt"
        val file = File(fileLocation)

        file.forEachLine { fileLines.add(it) }

        return fileLines

    }
}

File: python/src/main/python/utils/text_file_processor.py
class TextFileProcessor:
    def __init__(self, year: int, day_number: int) -> None:
        self.__year = year
        self.__day_number = day_number

    def get_file_lines(self) -> list[str]:
        file_location: str = (f"C:\\Users\\Kane Mullett\\advent-of-code\\python\\src\\main\\resources"
                              f"\\aoc{self.__year}\\day{self.__day_number}.txt")
        file = open(file_location, "r")

        return file.readlines()
